### PR TITLE
feat(platform/gitlab): handle detailed_merge_status to proceed with automerge

### PR DESCRIPTION
**Pull request by javaxiss**
## Changes

The behavior changed by this PR concerns only the GitLab platform. Before proceeding with automerge, a check is done on GitLab platform version and Renovate will rely on `detailed_merge_status` instead of `merge_status` attribute only if GitLab platform version is ^15.6.0.

RENOVATE_X_GITLAB_MERGE_REQUEST_DELAY environment variable is introduced to allow users define their own delay value, it is initialized to 250 by default.

## Context

This PR is a follow-up to the discussion [#25433](https://github.com/renovatebot/renovate/discussions/25433). It is a version feature gate on Gitlab platform (^15.6.0) : To proceed with `automerge` Renovate will check `detailed_merge_status` instead of `merge_status`, which is [deprecated in Gitlab 15.6](https://gitlab.com/gitlab-org/gitlab/-/issues/3169#note_1162532204)

Prior to Gitlab 15.6, Renovate continues to rely on `merge_status` attribute.
## Documentation (please check one with an [x])

- [X] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [X] Code inspection only, or
- [X] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/gitlab/index.ts [hunk 1]
```diff
@@ -644,26 +644,46 @@ async function tryPrAutomerge(
...
-      const desiredStatus = 'can_be_merged';
+      let desiredStatus = 'can_be_merged';
       // The default value of 5 attempts results in max. 13.75 seconds timeout if no pipeline created.
       const retryTimes = parseInteger(
         process.env.RENOVATE_X_GITLAB_AUTO_MERGEABLE_CHECK_ATTEMPS,
         5,
       );
 
+      if (semver.gte(defaults.version, '15.6.0')) {
+        logger.debug(
```
Comment (viceice) [inline lib/modules/platform/gitlab/index.ts:655]: I think this can be trace level?

@rarkins WDYT?

File: lib/modules/platform/gitlab/index.ts [hunk 2]
```diff
@@ -644,26 +644,46 @@ async function tryPrAutomerge(
...
+      const mergeDelay = parseInteger(
+        process.env.RENOVATE_X_GITLAB_MERGE_REQUEST_DELAY,
+        250,
+      );
+
       // Check for correct merge request status before setting `merge_when_pipeline_succeeds` to  `true`.
       for (let attempt = 1; attempt <= retryTimes; attempt += 1) {
         const { body } = await gitlabApi.getJson<{
           merge_status: string;
+          detailed_merge_status: string;
```
Comment (viceice) [inline lib/modules/platform/gitlab/index.ts:671]:
```suggestion
          detailed_merge_status?: string;
```
can be undefined

**Review by viceice — COMMENTED** [review]
needs new tests for coverage

**Comment by viceice** [comment]
you need to fix git author and committee and then force push. otherwise we can't merge this

**Comment by javaxiss** [comment]
> you need to fix git author and committee and then force push. otherwise we can't merge this

done, I have taken onboard your suggestions and remarks, as well :relaxed:

**Review by viceice — COMMENTED** [review]
it seems you've still a wrong git author or committee, otherwise LGTM

File: lib/modules/platform/gitlab/index.spec.ts [hunk 3]
```diff
@@ -1824,6 +1824,49 @@ describe('modules/platform/gitlab/index', () => {
...
+        .get('/api/v4/projects/undefined/merge_requests/12345')
+        .reply(200)
+        .get('/api/v4/projects/undefined/merge_requests/12345')
+        .reply(200)
+        .get('/api/v4/projects/undefined/merge_requests/12345')
+        .reply(200)
+        .put('/api/v4/projects/undefined/merge_requests/12345/merge')
+        .reply(200);
+      process.env.RENOVATE_X_GITLAB_AUTO_MERGEABLE_CHECK_ATTEMPS = '3';
+      process.env.RENOVATE_X_GITLAB_MERGE_REQUEST_DELAY = '250';
```
Comment (viceice) [inline lib/modules/platform/gitlab/index.spec.ts:1846]: please delete those in `beforeEach`, there should be some samples.

you should also not test with default fallback values.

Comment (javaxiss) [reply lib/modules/platform/gitlab/index.spec.ts]: done

**Review by viceice — APPROVED** [review]

**Pull request merged**